terralab/utils.py

```diff
@@ -29,10 +23,55 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
         except ApiException as e:
             formatted_message = f"API call failed with status code {e.status} ({e.reason}): {json.loads(e.body)['message']}"
-            LOGGER.error(formatted_message)
+            LOGGER.error(add_blankline_after(formatted_message))
             exit(1)
         except Exception as e:
-            LOGGER.error(str(e))
+            LOGGER.error(add_blankline_after(str(e)))
             exit(1)
 
     return wrapper
+
+
+def process_json_to_dict(json_data) -> dict:
+    """Process the given JSON string and return a dictionary.
+    Returns None if the input string is not able to be parsed to a dictionary."""
+    try:
+        data = json.loads(json_data)
+        if not (isinstance(data, dict)):
+            raise TypeError
```

Comment:
Is this `raise` something we're doing with from the caller or should we log an `error` and exit here as well?

Reply:
good point, elsewhere we're logging and error and exiting, and we should be consistent

Reply:
if the message logged is some generic thing it would prob be better to log something more specific here instead

Reply:
it's in line 45: "Input string not parsable to a dictionary."